### DISCUSSION TITLE
Nodes do not join the cluster

### DISCUSSION THREAD

**Discussion by clickersmudge** [kubermatic/kubeone#3166] [Q&A] [answered]
Hi,
I have a KubeOne cluster v1.7.3

```
kubectl get machinedeployment -A
NAMESPACE     NAME              REPLICAS   AVAILABLE-REPLICAS   PROVIDER   OS       KUBELET   AGE
kube-system   prod-pool1        0                               hetzner    ubuntu   1.27.11   12d
kube-system   prod-worker-arm   1          1                    hetzner    ubuntu   1.27.11   12d
```

When it adds a new machinedeploymnet from a file that worked early on.
The machine is created at Hetzner.
![obraz](https://github.com/kubermatic/kubeone/assets/113194014/492212d3-7d72-4589-a161-18f77dba6524)

A new machinedeployment is created
![obraz](https://github.com/kubermatic/kubeone/assets/113194014/68e19670-7681-4ddf-90e0-2928b27982f3)

but a new node never appears
![obraz](https://github.com/kubermatic/kubeone/assets/113194014/9a7e1750-3046-4312-94a2-9d8f511f2ec8)

```
kubectl  describe machinedeployment -n kube-system <NAME>
```

Shows warning.
 **Warning  ReconcileError  2m42s (x3 over 2m42s)  machinedeployment-controller  Operation cannot be fulfilled on machinedeployments.cluster.k8s.io "prod-worker-x86": the object has been modified; please apply your changes to the latest version and try again**

I checked further
```
kubectl get machines --namespace kube-system
kubectl get machineset --namespace kube-system
```

I created a new MachineDeployment with a different name
![obraz](https://github.com/kubermatic/kubeone/assets/113194014/0cc384c3-199e-4ee6-822a-a5575ee20043)

The machine has also created.
![obraz](https://github.com/kubermatic/kubeone/assets/113194014/dea3d211-59b2-464e-b3a3-b9e6b7737056)

**After about 10-15 minutes, the old machine is deleted, a new one is created, but none joins the cluster.**

![obraz](https://github.com/kubermatic/kubeone/assets/113194014/ed0e6537-898f-4c45-86d5-4dda6c07dd52)

```
 kubectl logs -n kube-system                     machine-controller-7b47d7b45b-bhf5h
```
![obraz](https://github.com/kubermatic/kubeone/assets/113194014/38df02e9-536c-4cae-af75-bff6c65afb7e)

**Comment by kron4eg** [comment, marked as answer]
Hello!

Yeah, we are aware of the situation (see #3165) and already merged the fix, the releases is pending. For now you can fix this by deploying updating version of the OSM (see #3163 for 1.7 release).
